src/core.jl

```diff
@@ -321,14 +356,78 @@ end
 @inline _normalize(xf) = xf
 @inline _normalize(xf::Composition{<:Composition}) = xf.outer |> xf.inner
 
-# Not sure if this a good idea... (But it's easier to type)
-@inline Base.:|>(f::Composition, g::Transducer) = f.outer |> (f.inner |> g)
-@inline Base.:|>(f::Transducer, g::Transducer) = Composition(f, g)
-# Base.∘(f::Transducer, g::Transducer) = Composition(f, g)
-# Base.∘(f::Composition, g::Transducer) = f.outer ∘ (f.inner ∘ g)
-@inline Base.:|>(::IdentityTransducer, f::Transducer) = f
-@inline Base.:|>(f::Transducer, ::IdentityTransducer) = f
-@inline Base.:|>(f::Composition, ::IdentityTransducer) = f  # disambiguation
+"""
+    f ⨟ g
+    g ∘ f
+    opcompose(f, g)
+    compose(g, f)
+
+Composition of transducers.
+
+!!! compat "Transducers.jl 0.4.XX"
+
+    Transducers.jl 0.4.XX or later is required for composing
+    transducers with `∘` and other operators and functions derived
+    from it.
+
+    Transducers written as `f |> g |> h` in previous versions of
+    Transducers.jl can now be written as `f ⨟ g ⨟ h` (in Julia 1.5 or
+    later) or `opcompose(f, g, h)`.
+
+!!! note
+
+    "op" in `opcompose` does not stand for _operator_; it stands for
+    _opposite_.
+"""
+@inline Base.:∘(g::Transducer, f::Transducer) = Composition(f, g)
+@inline Base.:∘(g::Transducer, f::Composition) = g ∘ f.inner ∘ f.outer
+@inline Base.:∘(f::Transducer, ::IdentityTransducer) = f
+@inline Base.:∘(::IdentityTransducer, f::Transducer) = f
```

Comment:
```suggestion
@inline Base.:∘(::IdentityTransducer, f::Transducer) = f
@inline Base.:∘(::IdentityTransducer, ::IdentityTransducer) = IdentityTransducer()
```
Should Aqua have caught this? There might be more signatures missing?

Reply:
Ah, thanks, good catch!  I'm not doing "full" test with Aqua at the moment since there are a quite lot of ambiguity (due to initial design flaw).  I'll see if I can enable it.